services/udp/service.go

```diff
@@ -98,11 +101,13 @@ func (s *Service) Open() (err error) {
 		s.Logger.Printf("Failed to set up UDP listener at address %s: %s", s.addr, err)
 		return err
 	}
+	s.conn.SetReadBuffer(s.config.ReadBuffer)
```

Comment:
This is probably the most significant change of this entire PR, increasing the read buffer size has a dramatic improvement in the ability to handle large bursts of UDP traffic.